### PR TITLE
rename psp20 to psp17, add ipsp17receiver

### PR DESCRIPTION
**Pull request by VargSupercolony**

### CODE REVIEW

File: PSPs/drafts/psp-17.md [hunk 1]
```diff
@@ -109,6 +112,15 @@ pub enum PSP20Error {
 }
 ```
 
+PSP17ReceiverError:
+
+```rust
+pub enum IPSP17ReceiverError {
+ /// Returned if a transfer is rejected.
+ TransferRejected(String),
```
Comment (xgreenx) [inline PSPs/drafts/psp-17.md:123]: &'static str

File: PSPs/drafts/psp-17.md [hunk 2]
```diff
@@ -56,13 +54,18 @@ pub trait IPSP20 {
...
+pub trait IPSP17Metadata {
  fn token_name(&self) -> Option<String>;
 
  fn token_symbol(&self) -> Option<String>;
 
  fn token_decimals(&self) -> u8;
 }
+
+pub trait IPSP17Receiver {
+ fn on_psp17_received(&mut self, operator: AccountId, from: AccountId, value: Balance, data: Vec<u8>) -> Result<(), PSP17ReceiverError>;
```
Comment (xgreenx) [inline PSPs/drafts/psp-17.md]: You need to describe behavior of this trait somewhere. For example like in solidity via comments

**Review by xgreenx — APPROVED** [review]

File: PSPs/drafts/psp-17.md [hunk 3]
```diff
@@ -91,11 +97,11 @@ struct Approval {
...
 ### Errors
-Suggested methods don't return `Result`. Instead, they panic.
+Suggested methods don't return `Result` (except `on_psp17_received`). Instead, they panic.
 This panic can contain one of the following messages:
 
 ```rust
-pub enum PSP20Error {
+pub enum PSP17Error {
  /// Unknown error type for cases if writer of traits added own restrictions
  Unknown(&'static str),
```
Comment (xgreenx) [inline PSPs/drafts/psp-17.md]: String here too

**Review by xgreenx — CHANGES_REQUESTED** [review]

**Review by xgreenx — APPROVED** [review]

**Review by PierreOssun — APPROVED** [review]
Ok for me

File: PSPs/drafts/psp-17.md [hunk 4]
```diff
@@ -91,13 +95,13 @@ struct Approval {
...
-Suggested methods don't return `Result`. Instead, they panic.
+Suggested methods don't return `Result` (except `on_received`). Instead, they panic.
 This panic can contain one of the following messages:
 
 ```rust
-pub enum PSP20Error {
+pub enum PSP17Error {
  /// Unknown error type for cases if writer of traits added own restrictions
- Unknown(&'static str),
+ Unknown(String),
```
Comment (sventime) [inline PSPs/drafts/psp-17.md:104]: Let's maybe call it Custom. It substrate they will do the same for dymanic errors: https://github.com/paritytech/substrate/issues/5239#issuecomment-858451861

**Pull request merged**